web/app/habit/stake/page.tsx

```diff
@@ -159,28 +231,55 @@ export default function Join() {
           />
         </div>
 
+        {/**
+         * Enable only when challenge is selected
+         * If doesn't support batch tx and doesn't have enough balance -> Disable Button
+         * If doesn't support batch tx and has enough balance and no allowance -> Display approve && Approve Tx
+         * If doesn't support batch tx and has enough balance and has allowance -> Display stake && Stake Tx
+         * If support batch tx -> Display stake && Join with Batch Tx
+         */}
         <button
           type="button"
           className="bg-yellow bg-yellow mt-4 rounded-lg border-solid px-6 py-3 font-bold disabled:cursor-not-allowed disabled:bg-gray-400"
           style={{ width: '250px', height: '45px', color: 'white' }}
-          onClick={onJoinButtonClick}
-          disabled={!hasEnoughBalance || joinPending || isLoading}
+          onClick={hasEnoughAllowance ? onJoinButtonClick : onApproveTestTokenClick}
+          disabled={
+            !selectedChallenge ||
+            (!hasEnoughBalance && !currentChainSupportBatchTx) ||
+            mintPending ||
+            approvePending ||
+            joinPending ||
+            joinPendingInBatchTx ||
+            isLoading
+          }
         >
-          {selectedChallenge ? `Stake ${selectedChallenge.stake} ETH` : 'Stake'}
+          {selectedChallenge === null
+            ? 'Choose a Challenge'
+            : hasEnoughAllowance
+            ? `Stake ${selectedChallenge.stake} ALI`
+            : 'Approve'}
         </button>
 
+        {/**
+         * Display only when challenge is selected
+         * If doesn't support batch tx and doesn't have enough balance -> Mint first
+         * If doesn't support batch tx and has enough balance -> Show balance
+         * If support batch tx -> Show balance
+         */}
         <div className="p-4 text-xs">
-          {balance.data && hasEnoughBalance ? (
-            <p> 💰 Smart Wallet Balance: {ethBalance.toString()} ETH </p>
-          ) : (
+          {!selectedChallenge ? (
+            <p> </p>
+          ) : balance.data && !currentChainSupportBatchTx && !hasEnoughBalance ? (
```

Comment:
```suggestion
          ) : balance.data && !hasEnoughBalance ? (
```

If user doesn't have enough, mint and ignore batching